### PR TITLE
Added initial Boyle/Coello soiling model

### LINKED ISSUE #25: Feature Request: soiling model
A soiling model was presented by Merissa Coello and Liza Boyle from Humboldt University at PVSC/WCPEC 2018. Requires time-series data of particulate matter (PM) concentrations for 10 micron and 2.5 micron particle diameters, and rainfall. Other parameters needed for variable deposition models. I have a copy of the model paper from Merissa and Liza, but can't find a reference to the proceedings. I have a model function developed and am putting final touches on the model. I'll PR when complete.

### PR DESCRIPTION
**Pull request by DanRiley**
Closes #25.

### CODE REVIEW

**Comment by DanRiley** [comment]
I thought I should let you know, @Wholmgren, that I'd implemented a new soiling model in MATLAB.

**Comment by wholmgren** [comment]
Thanks @DanRiley!

File: pvl_hsu_soiling.m [hunk 1]
```diff
@@ -0,0 +1,351 @@
+function SR = pvl_hsu_soiling(Time, Rain, RainThresh, Tilt, PM2_5, PM10, varargin)
```
Comment (cwhanse) [inline pvl_hsu_soiling.m]: Reorder name to `pvl_soiling_hsu`  what does "hsu" stand for?

File: pvl_hsu_soiling.m [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+%      Uses the parameters of the fixed models along with Land Use Category
+%      (LUC), wind speed, and ambient temperature to predict soiling ratio
+%      using the variable deposition velocity model.
+%
+% Input Parameters:
+%   Time is a struct with elements as described in pvl_maketimestruct. 
+%     Time values for the soiling function do not need to be 
+%     regularly spaced, although large gaps in timing are discouraged.
+%
+%   Rain is a vector of rainfall values of the same length as the fields of
```
Comment (cwhanse) [inline pvl_hsu_soiling.m]: `Rain is a vector of rainfall in mm of the same length as Time

File: pvl_hsu_soiling.m [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+%     Time, where each element of Rain correlates with the element of Time.
+%     Rainfall values should be in mm of rainfall. Programmatically, rain
+%     is accumulated over a given time period, and cleaning is applied
+%     immediately after a time period where the cleaning threshold is
+%     reached.
+%
+%   RainThresh is a scalar for the amount of rain, in mm, in a an accumulation
```
Comment (cwhanse) [inline pvl_hsu_soiling.m]: `in a an`

File: pvl_hsu_soiling.m [hunk 4]
```diff
@@ -0,0 +1,351 @@
...
+%     to be cleaned immediately after the accumulation period 
+%     [1] suggests a good RainThresh could be 1mm, but the time period is
+%     not specified. Rain accumulation period length can be adjusted in the
+%     optional input RainAccPeriod.
+%
+%   Tilt is a scalar or vector for the tilt of the PV panels. Changing tilt
+%     angles (e.g. in tracking cases) can be accomodated, and tilt angles
+%     are correlated with the entries in Time.
+%
+%   PM2_5 is the concentration of particulate matter with diamter less than
```
Comment (cwhanse) [inline pvl_hsu_soiling.m]: airborne particular matter?  "diamter"

File: pvl_hsu_soiling.m [hunk 5]
```diff
@@ -0,0 +1,351 @@
...
+%     2.5 microns, in g/m^3. Historical US concentration data sets may be
+%     found at https://aws.epa.gov/aqsweb/airdata/download_files.html. With
+%     file descriptions (for hourly data sets) at 
+%     https://aqs.epa.gov/aqsweb/airdata/FileFormats.html#_hourly_data_files
+%   
+%   PM10 is the concentration of particulate matter with diamter less than
```
Comment (cwhanse) [inline pvl_hsu_soiling.m]: "diamter"

File: pvl_hsu_soiling.m [hunk 6]
```diff
@@ -0,0 +1,351 @@
...
+%     10 microns, in g/m^3. Historical US PM data sets found as described
```
Comment (cwhanse) [inline pvl_hsu_soiling.m]: what is "PM"

File: pvl_hsu_soiling.m [hunk 7]
```diff
@@ -0,0 +1,351 @@
...
+%     over which to accumulate rainfall totals before checking against the
+%     rain cleaning threshold. For example, if the rain threshold is
+%     0.5 mm per hour, then RainThresh should be 0.5 and RainAccPeriod
+%     should be 1. If the threshold is 1 mm per hour, then the values
+%     should be 1 and 1, respectively. The minimum RainAccPeriod is 1
+%     hour. The default value is 1, indicating hourly rain accumulation.
+%     Accumulation periods exceeding 24 (daily accumulation) are not
+%     recommended.
+%   
+%   LUC is an optional input to the function, but it is required for the
```
Comment (cwhanse) [inline pvl_hsu_soiling.m]: LUC is a required input if the Variable Deposition Model is selected, i.e., ModelType = 1

File: pvl_hsu_soiling.m [hunk 8]
```diff
@@ -0,0 +1,351 @@
...
+%     Variable Deposition Model. LUC is the Land Use Category as specified
+%     in Table 19.2 of [2]. LUC must be a numeric scalar with value 1, 4,
+%     6, 8, or 10, corresponding to land with evergreen trees, deciduous
+%     trees, grass, desert, or shrubs with interrupted woodlands. The LUC
```
Comment (cwhanse) [inline pvl_hsu_soiling.m]: I'd delete the sentence about Brownian diffusion and the text about gamma (which isn't an input?).  It's not enough to explain the model, and adding more would make the variable definition too long.  If you want to include the material I'd make it comments in the function.

File: pvl_hsu_soiling.m [hunk 9]
```diff
@@ -0,0 +1,351 @@
...
+%     changes the mass deposition by Brownian diffusion. If omitted, the
+%     default value of 8 (desert) is used. Note that the value of gamma is
+%     the only thing changed by the LUC. LUC 1 and 4 use gamma = 0.54,
+%     while LUC 6, 8, and 10 use gamma = 0.56.
+%
+%   WindSpeed is an optional input to the function, but is required for the
```
Comment (cwhanse) [inline pvl_hsu_soiling.m]: WindSpeed is a required input if the Variable Deposition Model is selected, i.e., ModelType = 1

File: pvl_hsu_soiling.m [hunk 10]
```diff
@@ -0,0 +1,351 @@
...
+%     Variable Deposition Model. WindSpeed is a scalar or vector value with
+%     the same number of elements as Time, and must be in meters per
+%     second. If WindSpeed is omitted, the value of 2 m/s is used as
+%     default.
+%
+%   Temperature is an optional input to the function, but is required for
```
Comment (cwhanse) [inline pvl_hsu_soiling.m]: Temperature is a required input if the Variable Deposition Model is selected, i.e., ModelType = 1

File: pvl_hsu_soiling.m [hunk 11]
```diff
@@ -0,0 +1,351 @@
...
+%     the Variable Deposition Model. Temperature is a scalar or vector
+%     value with the same number of Elements as Time and must be in degrees
+%     C. If Temperature is omitted, the value of 12 C is used as default.
+% 
+% 
+% Output Parameters:
+%   SR = The soiling ratio (SR) of a tilted PV panel, this is a number
```
Comment (cwhanse) [inline pvl_hsu_soiling.m]: SR is a time series?

File: pvl_hsu_soiling.m [hunk 12]
```diff
@@ -0,0 +1,351 @@
...
+SoilingRate = 34.37 * erf(0.17 .* (TiltedMass.^0.8473));
+
+% Soiling Ratio is the % soil transmittance (0.95 = 5% soiling loss)
+SR = (100 - SoilingRate) ./ 100;
+
+end
+
+% This function creates deposition velocities for the variable deposition
+% model.
+function vd = pvl_depo_veloc(T, WindSpeed, LUC)
```
Comment (cwhanse) [inline pvl_hsu_soiling.m]: Minor style point - I'd rather not use the prefix "pvl" for private functions.

File: pvl_hsu_soiling.m [hunk 13]
```diff
@@ -0,0 +1,351 @@
...
+            gamma = 0.56;
+        case {6, 8, 10}
+            gamma = 0.54;
+        otherwise
+            warning('Unknown Land Use Category (LUC), assuming LUC 8.');
+            gamma = 0.54;
+    end
+
+
+    %Dimeter of particle in um
```
Comment (cwhanse) [inline pvl_hsu_soiling.m]: Diameter

File: pvl_hsu_soiling.m [hunk 14]
```diff
@@ -0,0 +1,351 @@
...
+    Dpum=[2.5,10];
+    Dpm=Dpum*10^-6;   %Diameter of particle in m
+
+    %Calculations
+    mu=1.8*10^-5.*(T./298).^0.85;      %viscocity of air in kg/m s
```
Comment (cwhanse) [inline pvl_hsu_soiling.m]: viscosity

**Review by cwhanse — CHANGES_REQUESTED** [review]
Edits to signature and some typos.

File: pvl_hsu_soiling.m (hunk 1 above)
Comment (DanRiley) [reply pvl_hsu_soiling.m]: hsu = Humboldt State University. I asked Liza if she had a preferred name for the model (e.g. Boyle-Coello), and she said that the given name "pvl_hsu_soiling" was acceptable. 

I am open to changes, though, this was really just my first go at naming the model.

I'll change the file and function name as requested.

Comment (cwhanse) [reply pvl_hsu_soiling.m]: "hsu" it is, would be nice to find a way to work Humboldt state University into the notes.

File: pvl_hsu_soiling.m (hunk 11 above)
Comment (DanRiley) [reply pvl_hsu_soiling.m]: Corrected as follows:
%   SR = The soiling ratio (SR) of a tilted PV panel, this is a number
%   between 0 and 1. SR is a time series where each element of SR
%   correlates with the accumulated soiling and rain cleaning at the times
%   specified in Time.

File: pvl_hsu_soiling.m (hunk 5 above)
Comment (DanRiley) [reply pvl_hsu_soiling.m]: corrected

File: pvl_hsu_soiling.m (hunk 6 above)
Comment (DanRiley) [reply pvl_hsu_soiling.m]: Corrected to indicate that PM is the abbreviation for Particulate Matter.

File: pvl_hsu_soiling.m (hunk 4 above)
Comment (DanRiley) [reply pvl_hsu_soiling.m]: Corrected.

File: pvl_hsu_soiling.m (hunk 3 above)
Comment (DanRiley) [reply pvl_hsu_soiling.m]: Corrected

File: pvl_hsu_soiling.m (hunk 2 above)
Comment (DanRiley) [reply pvl_hsu_soiling.m]: Corrected.

File: pvl_hsu_soiling.m (hunk 8 above)
Comment (DanRiley) [reply pvl_hsu_soiling.m]: Removed the two sentences about Brownian diffusion and gamma without adding comments in the function. Users should reference Seinfeld and Pandis for more information.

File: pvl_hsu_soiling.m (hunk 14 above)
Comment (DanRiley) [reply pvl_hsu_soiling.m]: Corrected

File: pvl_hsu_soiling.m (hunk 13 above)
Comment (DanRiley) [reply pvl_hsu_soiling.m]: Corrected

File: pvl_hsu_soiling.m (hunk 12 above)
Comment (DanRiley) [reply pvl_hsu_soiling.m]: Corrected, new name is "depo_veloc".

File: pvl_hsu_soiling.m (hunk 7 above)
Comment (DanRiley) [reply pvl_hsu_soiling.m]: It's optional in that the function will generate an error if it is omitted. In the event it is omitted, a default value of 8 (desert) is used if LUC is omitted. 

Whether we call this "optional" or "required" is perhaps a matter of semantics that I think is resolved in the full text description.

Similar response to the other "optional/required" variables of Ambient Temperature and Wind Speed.

File: pvl_hsu_soiling.m (hunk 1 above)
Comment (DanRiley) [reply pvl_hsu_soiling.m]: Really great point, I don't think I've given them appropriate credit. I added a whole section between the Output Parameters and the References:
% Background and recognition:
%   This soiling model was developed by Liza Boyle and Merissa Coello at
%   Humboldt State University (HSU). Many thanks to Liza and Merissa for
%   model development, initial code, and subsequent code and performance
%   reviews.

**Review by cwhanse — APPROVED** [review]
Ready to merge @DanRiley ?

**Comment by DanRiley** [comment]
I think so. After updating the reference to the IEEE JPV, I don't have any other changes.

**Pull request merged**